packages/core/src/api/blockManipulation/blockManipulation.ts

```diff
@@ -60,13 +66,31 @@ export function insertBlocks<
         ttEditor.state.tr.insert(insertionPos, blockGroupNode)
       );
 
-      return;
+      insertionPos++;
+    } else {
+      insertionPos = posBeforeNode + node.firstChild!.nodeSize + 2;
     }
+  }
 
-    insertionPos = posBeforeNode + node.firstChild!.nodeSize + 2;
+  const insertedBlocks: Block<BSchema, I, S>[] = [];
```

Comment:
why can't we just return `nodesToInsert`? I have some guesses (i.e.: ids), but from a code readability / maintainability perspective it seems weird ("future-us" would ask: "why do we query the nodes again"?

So I think it's smart to explain this in comments

Reply:
Ah this is just an oversight from my part - forgot that nodes are assigned IDs before being added to the editor, so this is not needed.